src/wrappers/foresight/lib/handlers/Setup.ts

```diff
@@ -0,0 +1,121 @@
+import * as TestRunnerSupport from '../../TestRunnerSupport';
+import ForesightWrapperUtils from '../../ForesightWrapperUtils';
+import * as EnvironmentSupport from '../../environment/EnvironmentSupport';
+import ExecutionContextManager from '../../../../context/ExecutionContextManager';
+import TestSuiteExecutionContext from '../../model/TestSuiteExecutionContext';
+import TestSuiteEvent from '../../model/TestSuiteEvent';
+import Utils from '../../../../utils/Utils';
+import ConfigProvider from '../../../../config/ConfigProvider';
+import ThundraLogger from '../../../../ThundraLogger';
+
+async function sendData(data: any) {
+
+    const config = ConfigProvider.thundraConfig;
+    const { apiKey } = config;
+
+    const { reporter } = TestRunnerSupport.wrapperContext;
+
+    await reporter.sendReports([Utils.generateReport(data, apiKey)]);
+}
+
+async function globalSetup() {
+
+    ThundraLogger.debug(`<Setup> Global setup creating for test suite: ${TestRunnerSupport.testSuiteName}.`);
+
+    await EnvironmentSupport.init();
+
+    const testRunStart = TestRunnerSupport.startTestRun();
+    if (!testRunStart) {
+        return;
+    }
+
+    try {
+
+        await sendData(testRunStart);
+        ThundraLogger.debug(`
+            <Setup> Test run start event sended for test suite: ${TestRunnerSupport.testSuiteName}
+            with test run id: ${testRunStart.id}
+        `);
+        TestRunnerSupport.startTestRunStatusEvent();
```

Comment:
I think, we should start sending testrun status events even though testrun start event fails